modyn/trainer_server/internal/trainer/pytorch_trainer.py

```diff
@@ -362,16 +376,162 @@ def train(self) -> None:  # pylint: disable=too-many-locals, too-many-branches
         final_checkpoint_file_name = self._final_checkpoint_path / "model_final.modyn"
         self.save_state(final_checkpoint_file_name)
 
+        # clean temporary directories in dataloader
+        self.end_of_trigger_cleaning()
+
         self._info("Training complete!")
 
+    def downsample_batch(
+        self, data: torch.Tensor, sample_ids: list, target: torch.Tensor
```

Comment:
can you move this to the implementation for stb (so we have a semantic grouping of logic within the file) + add a short comment just like for ` downsample_trigger_training_set`?

Reply:
Yep, done